### PR TITLE
don’t doctest in extern by default

### PR DESCRIPTION
**Pull request by nstarman**
It's not meant to be safe by default. If it is, then this can be removed by the package author.

fixes: #486 

Signed-off-by: Nathaniel Starkman <nstarkman@protonmail.com>

### CODE REVIEW

**Comment by nstarman** [comment]
@astrofrog wrt #121, what do you think?

File: {{ cookiecutter.package_name }}/setup.cfg [hunk 1]
```diff
@@ -47,6 +47,7 @@ testpaths = "{{ cookiecutter.module_name }}" "docs"
 astropy_header = true
 doctest_plus = enabled
 text_file_format = rst
+doctest_norecursedirs=*/{{ cookiecutter.module_name }}/extern/*
```
Comment (bsipocz) [inline {{ cookiecutter.package_name }}/setup.cfg:50]: shound't this be 
```suggestion
doctest_norecursedirs={{ cookiecutter.module_name }}/extern/*
```

**Comment by bsipocz** [comment]
Would be nice to add some testing for all these new things, but I'm not sure how to do that nicely without littering the user rendered template. Maybe a non working code example in the `__init__.py` is enough to say it's not tested by default? (I'm not sure that file would be normally picked up for testing anyway)

**Comment by nstarman** [comment]
I'm noticing that astropy has a different setup for not recursing directories. Is there this a preferable design philosophy?

```
norecursedirs =
    ...
    "astropy[\/]extern"
```

**Comment by nstarman** [comment]
gentle ping @bsipocz.
Otherwise this PR is ready for review.

**Comment by pllim** [comment]
See https://github.com/astropy/package-template#deprecation-warning

**Pull request closed without merging**